docs/content/documentation/systems/Kernels/index.md

```diff
@@ -1,7 +1,160 @@
 <!-- MOOSE Documentation Stub: Remove this when content is added. -->
 
 # Kernels System
-!parameters /Kernels
+
+A "Kernel" is a piece of physics. It can represent one or more operators or
+terms in the weak form of a partial differential equation.  With all terms on
+the left-hand-side, their sum is referred to as the "residual". The residual
+is evaluated at several integration quadrature points over the problem
+mesh/geometry. To implement your own physics in MOOSE, you create your own
+kernel by subclassing the MOOSE `Kernel` class.  In a `Kernel` subclass:
+
+The `computeQpResidual()` function MUST be overridden.  This is where you
+implement your PDE weak form terms.  The following member functions can
+optionally be overriden/provided:
+
+* `computeQpJacobian()`
+* `computeQpOffDiagJacobian()`
+
+These two functions provide extra information that can help the numerical
+solver(s) converge faster and better.  Inside your Kernel class, you have
+access to several member variables for computing the residual and jacobian
+values in the above mentioned functions:
+
+* `_i`, `_j`: indices for the current test and trial shape functions respectively.
+* `_qp`: current quadrature point index.
+* `_u`, `_grad_u`: value and gradient of the variable this Kernel operates on;
+  indexed by `_qp` (i.e. `_u[_qp]`).
+* `_test`, `_grad_test`: value ($\psi$) and gradient ($\nabla \psi$) of the
+  test functions at the q-points; indexed by `_i` and then `_qp` (i.e. ``_test[_i][_qp]``).
+* `_phi`, `_grad_phi`: value ($\phi$) and gradient ($\nabla \phi$) of the
+    trial functions at the q-points; indexed by `_j` and then `_qp` (i.e. ``_phi[_j][_qp]``).
+* `_q_point`: XYZ coordinates of the current quadrature point.
+* `_current_elem`: pointer to the current element being operated on.
+
+## Custom Kernel Creation
+
+To create a custom kernel, you can follow the pattern of the `Diffusion`
+kernel implemented and included in the MOOSe framework.  The strong-form of
+the diffusion equations is defined as:
+
+$$
+\begin{aligned}
+-\nabla\cdot\nabla u - f &= 0\\
+u|_{\partial\Omega_1} &= g_1\\
+\nabla u\cdot \hat{n} |_{\partial\Omega_2} &= g_2\end{aligned}
+$$
+
```

Comment:
where $\partial\Omega_1$ is defined as the boundary on which the value of $u$ is fixed to a known constant $g_1$ and $\partial\Omega_2$ is defined as the boundary on which the flux across the boundary is fixed to a known constant $g_2$.